packages/elements/src/photon-medication-dropdown-full-width/photon-medication-dropdown-full-width.tsx

```diff
@@ -0,0 +1,490 @@
+
+//Virtual List
+import { createVirtualizer, VirtualItem } from '@tanstack/solid-virtual';
+
+// Disable the default dropdown animation
+// TODO: Remove when the dropdown is swapped for a regular list
+setDefaultAnimation('dropdown.show', null);
+
+interface DataItem<T> {
+  data: T;
+  // TODO: setting this to scroll to the correct index
+  // after reopening the list, but not working ATM
+  allItemsIdx: number;
+}
+
+interface GroupTitle {
+  title: string;
+}
+
+// Typescript and solid are annoying
+// eslint-disable-next-line @typescript-eslint/ban-types
+type ThisisNotAFunction<T> = Exclude<T, Function>;
+
+export const PhotonMedicationDropdownFullWidth = <
+  T extends { id: string } | { medicationId: string }
+>(props: {
+  data: Array<Treatment | PrescriptionTemplate | TreatmentOption>;
+  label?: string;
+  required: boolean;
+  placeholder?: string;
+  forceLabelSize?: boolean;
+  invalid?: boolean;
+  onSearchChange?: (search: string) => void;
+  displayAccessor: (selected: T, groupDisplay: boolean) => string | JSXElement;
+  disabled?: boolean;
+  onOpen?: () => void;
+  onHide?: () => void;
+  isLoading: boolean;
+  hasMore: boolean;
+  noDataMsg?: string;
+  helpText?: string;
+  fetchMore?: () => void;
+  selectedData?: T | undefined;
+  groups?: Array<{
+    label: string;
+    filter: (arr: T) => boolean | undefined;
+  }>;
+  showOverflow?: boolean;
+  optional?: boolean;
+  clearable?: boolean;
+  actionRef?: any;
+  open: boolean;
+  searchText: string;
+  onClose: () => void;
+}) => {
+  //refs
+  let ref: any;
+  let dropdownRef: any;
+  let listRef: HTMLDivElement | undefined;
+  let inputRef: any;
+  let overlayRef: HTMLDivElement | undefined;
+
+  if (props.actionRef) {
+    props.actionRef['clear'] = () => {
+      setSelected(undefined);
+      setSelectedIndex(-1);
+      inputRef.value = '';
+      debounceSearch('');
+      dispatchDeselect();
+    };
+  }
+
+  //signals
+  const [selected, setSelected] = createSignal<T | undefined>(undefined);
+  const [selectedIndex, setSelectedIndex] = createSignal(-1);
+
+  const debounceSearch = debounce(async (s: string) => {
+    if (props.onSearchChange) {
+      await props.onSearchChange(s);
+    }
+  }, 250);
+
+  createEffect(() => {
+    if (props.open) {
+      dropdownRef?.show();
+      handleDropdownShow();
+    }
+  });
+
+  createEffect(() => {
+    if (props.data.length === 0) {
+      // if the source data changes then reset the selected value
+      setSelected(undefined);
+    }
+  });
+
+  const showHelpText = (invalid: boolean) => {
+    if (invalid) {
+      if (props.helpText) {
+        return props.helpText;
+      }
+      if (props.required) {
+        return 'Please select data...';
+      }
+      return undefined;
+    } else {
+      return undefined;
+    }
+  };
+
+  const dispatchSelect = (datum: T) => {
+    const event = new CustomEvent('photon-data-selected', {
```

Comment:
should this be more specific?

Reply:
if I changed, I'd have to handle data selected and treatment selected in photon-medication-search because of photon-dropdown. I'll leave for now, so i don't have to define two event handlers that do the same thing.